### ISSUE TITLE
Default config fails when `php artisan config:cache` is called

### ISSUE THREAD

**Issue by jboonstra** [stechstudio/laravel-jwt#5]
The default configuration provided by `laravel-jwt` is not cacheable by a standard `php artisan config:cache` command. The problem appears to be that the `Lcobucci\JWT\Encoding\ChainedFormatter` class does not implement (nor do any of its constructor dependencies) the magic `__set_state()` method which is necessary to properly serialize/unserialize objects (so that they can be written to a static cache file).

To reproduce, create a new laravel project, confirm that `php artisan config:cache` works properly, then install `laravel-jwt` and attempt to `php artisan config:cache`, which should produce the error:

```
# create a new laravel project (v10.44.0 at the time of this writing)
composer create-project laravel/laravel laravel-jwt-test
cd laravel-jwt-test

# confirm that stock configuration is cacheable
php artisan config:cache

# install laravel-jwt (2.2 at the time of this writing)
composer require stechstudio/laravel-jwt

# this will produce a "Your configuration files are not serializable." error
php artisan config:cache
```

Here's the full output of the error from the final `config:cache` command:

```
> php artisan config:cache

   LogicException

  Your configuration files are not serializable.

  at vendor/laravel/framework/src/Illuminate/Foundation/Console/ConfigCacheCommand.php:73
     69▕             require $configPath;
     70▕         } catch (Throwable $e) {
     71▕             $this->files->delete($configPath);
     72▕
  ➜  73▕             throw new LogicException('Your configuration files are not serializable.', 0, $e);
     74▕         }
     75▕
     76▕         $this->components->info('Configuration cached successfully.');
     77▕     }

  1   bootstrap/cache/config.php:849
      Error::("Call to undefined method Lcobucci\JWT\Encoding\ChainedFormatter::__set_state()")
      +13 vendor frames

  15  artisan:35
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

If I temporarily disable the removal of `bootstrap/cache/config.php` that normally happens when an exception is thrown, and examine it, I can see that this is how the default configuration is being cached (I'm including only the portion relevant to this package/error):

```
  'jwt' =>
  array (
    'key' => 'base64:[snipped]',
    'signer' => 'Lcobucci\\JWT\\Signer\\Hmac\\Sha256',
    'lifetime' => 600,
    'audience' => 'laravel',
    'issuer' => 'laravel',
    'merge' => true,
    'validate' =>
    array (
      'audience' => true,
    ),
    'chained_formatter' =>
    \Lcobucci\JWT\Encoding\ChainedFormatter::__set_state(array(
       'formatters' =>
      array (
        0 =>
        \Lcobucci\JWT\Encoding\UnifyAudience::__set_state(array(
        )),
        1 =>
        \Lcobucci\JWT\Encoding\MicrosecondBasedDateConversion::__set_state(array(
        )),
      ),
    )),
  ),
```

The `__set_state()` method is being called to allow serializing the default config from `jwt/config.php`, but since the `lcobucci/jwt` package doesn't implement that method for any of the relevant classes (`ChainedFormatter`, `UnifyAudience`, and `MicrosecondBasedDateConversion`), the config can't be cached.

Presumably a fix might be to either move `ChainedFormatter` out of the config (looks to have been added in 5162a403) OR to write an implementation or wrapper of those classes that allows for serialization. The root classes are implemented as `final` so unfortunately they can't just be extended; another fix would be to submit a patch back to the `lcobucci/jwt` package to implement serialization for those classes.

Our fix for the moment is to refactor and use `lcobucci/jwt` directly since we don't necessarily need all of the things that `laravel-jwt` provides, but I figured I'd mention this issue since it might prevent others from using this package in a production context (where config would typically be cached).

**Comment by jszobody** [comment]
Thanks @jboonstra it sounds like I need to move that class out of the config and use string values there. That seems like it would align more with standard Laravel config file conventions. The string key can be wired back to the class in a service provider.

I'd gladly accept a PR for this if you feel up for it. Or if you've already moved on (sounds like you have) I'll get to this when I get a chance.

**Comment by jszobody** [comment]
@jboonstra I just tagged a fix in v2.2.1. Thanks again.

**Comment by jboonstra** [comment]
Thanks much for the quick fix - your PR was better than one I'd have put up!
